File: ios/CounterViewManager.m
//
//  CounterViewManager.m
//  CounterApp
//
//  Created by actdigital on 13/03/21.
//

#import <Foundation/Foundation.h>
#import "React/RCTViewManager.h"

@interface RCT_EXTERN_MODULE(CounterViewManager, RCTViewManager)
  RCT_EXPORT_VIEW_PROPERTY(count, NSNumber)
  RCT_EXPORT_VIEW_PROPERTY(onUpdate, RCTDirectEventBlock)
  RCT_EXTERN_METHOD(updateFromManager:(nonnull NSNumber *)node
                    count:(nonnull NSNumber *)count)
@end

File: ios/CounterViewManager.swift
//
//  CounterViewManager.swift
//  CounterApp
//
//  Created by actdigital on 13/03/21.
//

import Foundation

@objc(CounterViewManager)
class CounterViewManager: RCTViewManager {
  override func view() -> UIView! {
    return CounterView()
  }
  
  override static func requiresMainQueueSetup() -> Bool {
    return true
  }
  
  @objc func updateFromManager(_ node: NSNumber, count: NSNumber){
    DispatchQueue.main.async {
      let component = self.bridge.uiManager.view(forReactTag: node) as! CounterView
      component.update(value: count)
    }
  }

  
}

